src/main/java/io/druid/segment/loading/DataSegmentPullerURI.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 Metamarkets Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.segment.loading;
+
+import com.google.common.base.Predicate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+/**
+ * A DataSegmentPullerURI extends the normal DataSegmentPuller to allow URI based handlings
+ */
+public interface DataSegmentPullerURI extends DataSegmentPuller
```

Comment:
I realized there is really no reason for this to extend DataSegmentPuller, so I'm experimenting with it being its own interface separately. Will put up commit soon that fixes this comment.